### PR TITLE
adding optional variables 'zone' and 'ultra_ssd_enabled' which are required to support ultrassd managed drives

### PR DESCRIPTION
**Pull request by waylew-lexis**
- added optional variables 'zone' and 'ultra_ssd_enabled' which are required to support ultra ssd managed drives
- update terraform validation workflow (terraform.yml)
- added tflint.hcl

### CODE REVIEW

File: .tflint.hcl [hunk 1]
```diff
@@ -0,0 +1,5 @@
+plugin "azurerm" {
+  enabled = true
+  version = "0.14.0"
```
Comment (dutsmiller) [inline .tflint.hcl:3]: Update this to 1.x, we shouldn't be maintaining code this old.

File: main.tf [hunk 2]
```diff
@@ -136,14 +146,24 @@ resource "azurerm_windows_virtual_machine" "windows" {
     }
   }
 
+  boot_diagnostics {
```
Comment (dutsmiller) [inline main.tf:149]: duplicate entry from above.

File: main.tf [hunk 3]
```diff
@@ -95,12 +95,22 @@ resource "azurerm_linux_virtual_machine" "linux" {
...
```
Comment (dutsmiller) [inline main.tf:98]: This block should be conditionalized.

File: main.tf [hunk 4]
```diff
@@ -95,12 +95,22 @@ resource "azurerm_linux_virtual_machine" "linux" {
...
+    storage_account_uri = var.diagnostics_storage_account_uri
+  }
+
   os_disk {
     caching                   = var.operating_system_disk_cache
     storage_account_type      = var.operating_system_disk_type
     write_accelerator_enabled = var.operating_system_disk_write_accelerator
   }
 
+  additional_capabilities {
```
Comment (dutsmiller) [inline main.tf:108]: This block should probably be conditionalized

File: variables.tf [hunk 5]
```diff
@@ -1,196 +1,214 @@
...
+  default     = null
+}
+
+variable "ultra_ssd_enabled" {
+  description = "Should the capacity to enable Data Disks of the UltraSSD_LRS storage account type be supported on this Virtual Machine."
+  type        = bool
+  default     = false
+}
+
+variable "zone" {
```
Comment (dutsmiller) [inline variables.tf:186]: availability_zone is more descriptive

File: main.tf [hunk 6]
```diff
@@ -136,14 +146,24 @@ resource "azurerm_windows_virtual_machine" "windows" {
...
+    ultra_ssd_enabled = var.ultra_ssd_enabled
+  }
+
+  zone = var.zone
```
Comment (dutsmiller) [inline main.tf:163]: availability_zone is more descriptive

File: .tflint.hcl (hunk 1 above)
Comment (waylew-lexis) [reply .tflint.hcl:3]: that is for tflint version, not tf version https://github.com/terraform-linters/tflint-ruleset-azurerm

File: main.tf (hunk 2 above)
Comment (waylew-lexis) [reply main.tf]: needed for both linux and windows vm resource

File: main.tf (hunk 4 above)
Comment (waylew-lexis) [reply main.tf]: changed to a dynamic block

File: main.tf (hunk 6 above)
Comment (waylew-lexis) [reply main.tf]: changed to a dynamic block

**Pull request merged**